Jenkinsfile

```diff
@@ -15,62 +15,71 @@ pipeline {
-                                docker cp tests:/home/gradle/service/build .
-                            '''
+        stage('Tests') {
+            steps {
+                script {
+                    sh '''
+                        docker-compose -f docker/docker-compose.yml -f docker/docker-compose-tests.yml build
+                        docker-compose -f docker/docker-compose.yml -f docker/docker-compose-tests.yml up --exit-code-from gp2gp
+                        docker cp tests:/home/gradle/service/build .
+                    '''
 
-                            archiveArtifacts artifacts: 'build/reports/**/*.*', fingerprint: true
-                            junit '**/build/test-results/**/*.xml'
-                            recordIssues(
-                                enabledForFailure: true,
-                                tools: [
-                                    checkStyle(pattern: 'build/reports/checkstyle/*.xml'),
-                                    spotBugs(pattern: 'build/reports/spotbugs/*.xml')
-                                ]
-                            )
-                        }
-                    }
-                    post {
-                        always {
-                            step([
-                                $class : 'JacocoPublisher',
-                                execPattern : '**/build/jacoco/*.exec',
-                                classPattern : '**/build/classes/java',
-                                sourcePattern : 'src/main/java',
-                                exclusionPattern : '**/*Test.class'
-                            ])
-                            sh "rm -rf build"
-                            sh "docker-compose -f docker/docker-compose.yml -f docker/docker-compose-tests.yml down"
-                        }
-                    }
+                    archiveArtifacts artifacts: 'build/reports/**/*.*', fingerprint: true
+                    junit '**/build/test-results/**/*.xml'
+                    recordIssues(
+                        enabledForFailure: true,
+                        tools: [
+                            checkStyle(pattern: 'build/reports/checkstyle/*.xml'),
+                            spotBugs(pattern: 'build/reports/spotbugs/*.xml')
+                        ]
+                    )
                 }
+            }
+            post {
+                always {
+                    step([
+                        $class : 'JacocoPublisher',
+                        execPattern : '**/build/jacoco/*.exec',
+                        classPattern : '**/build/classes/java',
+                        sourcePattern : 'src/main/java',
+                        exclusionPattern : '**/*Test.class'
+                    ])
+            sh "rm -rf build"
+                    sh "docker-compose -f docker/docker-compose.yml -f docker/docker-compose-tests.yml down"
```

Comment:
It is worth to dump all container logs into artefacts

Reply:
main purpose of this PR is to:
* add stage where we will do vulnerability scanning and fail pipeline if there are any which can be fixed
* fix every vulnerability in container which is fixable
* remove blank/unused "Build" stage

feel free to raise another PR with your proposal @bartek-sarul